pkg/nuclide-code-format/lib/CodeFormatManager.js

```diff
@@ -62,7 +63,11 @@ class CodeFormatManager {
     }
 
     const provider = matchingProviders[0];
-    if (provider.formatCode != null) {
+    // If formatCode exists, use it if selection isn't empty, or if it's empty
+    // but formatEntireFile doesn't exist.
+    if (provider.formatCode != null &&
+      (!selectionRangeEmpty ||
+        (selectionRangeEmpty && provider.formatEntireFile == null))) {
```

Comment:
what do you mean? the second "selectionRangeEmpty" check is redundant, so it actually avoids an extra check - to clarify, what I mean is

```
if (provider.formatCode != null && (!selectionRangeEmpty || provider.formatEntireFile == null))) {
```

IMO it's just as readable - the explanation would be "use me if the range is non-empty, or we're forced to because `formatEntireFile` isn't defined."